plugins/signals-backend/src/service/SignalManager.ts

```diff
@@ -64,6 +68,32 @@ export class SignalManager {
       onEvent: (params: EventParams) =>
         this.onEventBrokerEvent(params.eventPayload as SignalPayload),
     });
+
+    this.pingInterval = setInterval(() => this.ping(), 30000);
+  }
+
+  private ping() {
+    this.connections.forEach(conn => {
+      if (!conn.isAlive) {
+        this.logger.debug(`Connection ${conn.id} is not alive, terminating`);
+        conn.ws.terminate();
+        return;
+      }
+
+      conn.isAlive = false;
+      conn.ws.ping();
+    });
+  }
+
+  onServerClose() {
+    if (this.pingInterval) {
+      clearInterval(this.pingInterval);
+    }
+
+    this.connections.forEach(conn => {
+      conn.ws.terminate();
```

Comment:
Don't think this is gonna do much when the server is already closed right?

Reply:
Hm, true, but hooking this up with lifecycle service might be good idea